creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -584,17 +716,23 @@ class SnapbackSM {
         throw new Error('processStateMachineOperation():issueUpdateReplicaSetOp() Error')
       }
 
-      // Increment and adjust current slice by ModuloBase
+      // Increment and adjust current slice by this.moduloBase
       const previousModuloSlice = this.currentModuloSlice
       this.currentModuloSlice += 1
-      this.currentModuloSlice = this.currentModuloSlice % ModuloBase
+      this.currentModuloSlice = this.currentModuloSlice % this.moduloBase
+
+      // Clear this.healthyNodes for next modulo slice
+      this.healthyNodes = null
+
+      // Reinitialize this.endpointToSpIdMap to maintain freshness
+      await this.updateEndpointToSpIdMap()
```

Comment:
per above comments, clearing healthyNodes and refreshing `updateEndpoitToSpIdMap()` feel a bit out of place here. rather than setting class var for both and refreshing after each snapback run, simpler to just fetch at start of each run and assign result to local var which you pass into processing logic

Reply:
pass in params

Reply:
keep as is with try/catch and backup logic